lib/modules/dosomething/dosomething_api/resources/signup_resource.inc

```diff
@@ -189,6 +189,25 @@ function _signup_resource_index($user, $users, $campaigns, $count, $page, $compe
     'runs' => $runs,
   ];
 
+  if (variable_get('rogue_collection', FALSE)) {
+    if ($user) {
+      $northstar_user_id = dosomething_user_get_northstar_id($user->uid);
+    }
+
+     $rogue = dosomething_rogue_client();
+
+     $response = $rogue->getActivity([
+       'filter' => [
+         'northstar_id' => $northstar_user_id,
+         'campaign_id' => $campaigns[0],
+       ],
+     ]);
+
+     if ($response) {
+      return $response;
```

Comment:
Nope nope, you'd be doing that inline here! Once it looks how you like, safe to just return it. 😄